hydromt/metadata_resolver/slippy_map_tile_resolver.py

```diff
@@ -0,0 +1,121 @@
+        overview level or tuple with resolution and unit
+    geom: gpd.GeoSeries, optional
+        geometry to determine res if zoom_level or source in degree
+    zls_dict: dict, optional
+        dictionary with overview levels and corresponding resolution
+    dst_crs: pyproj.CRS, optional
+        destination crs to determine res if zoom_level tuple is provided
+        with different unit than dst_crs
+    """
+    # check zoom level
+    if zls_dict is None or len(zls_dict) == 0 or zoom_level is None:
+        return None
+    elif isinstance(zoom_level, int):
+        if zoom_level not in zls_dict:
+            raise ValueError(
+                f"Zoom level {zoom_level} not defined." f"Select from {zls_dict}."
+            )
+        zl = zoom_level
+        dst_res = zls_dict[zoom_level]
+    elif (
+        isinstance(zoom_level, tuple)
+        and isinstance(zoom_level[0], (int, float))
+        and isinstance(zoom_level[1], str)
+        and len(zoom_level) == 2
+        and dst_crs is not None
+    ):
+        src_res, src_res_unit = zoom_level
+        # convert res if different unit than crs
+        dst_crs = CRS.from_user_input(dst_crs)
+        dst_crs_unit = dst_crs.axis_info[0].unit_name
+        dst_res = src_res
+        if dst_crs_unit != src_res_unit:
+            known_units = ["degree", "metre", "US survey foot", "meter", "foot"]
+            if src_res_unit not in known_units:
+                raise TypeError(
+                    f"zoom_level unit {src_res_unit} not understood;"
+                    f" should be one of {known_units}"
+                )
+            if dst_crs_unit not in known_units:
+                raise NotImplementedError(
+                    f"no conversion available for {src_res_unit} to {dst_crs_unit}"
+                )
+            conversions = {
+                "foot": 0.3048,
+                "metre": 1,  # official pyproj units
+                "US survey foot": 0.3048,  # official pyproj units
+            }  # to meter
+            if src_res_unit == "degree" or dst_crs_unit == "degree":
+                lat = 0
+                if geom is not None:
+                    lat = geom.to_crs(4326).centroid.y.item()
+                conversions["degree"] = cellres(lat=lat)[1]
+            fsrc = conversions.get(src_res_unit, 1)
+            fdst = conversions.get(dst_crs_unit, 1)
+            dst_res = src_res * fsrc / fdst
+        # find nearest zoom level
+        res = list(zls_dict.values())[0] / 2
+        zls = list(zls_dict.keys())
+        smaller = [x < (dst_res + res * 0.01) for x in zls_dict.values()]
+        zl = zls[-1] if all(smaller) else zls[max(smaller.index(False) - 1, 0)]
+    elif dst_crs is None:
+        raise ValueError("No CRS defined, hence no zoom level can be determined.")
+    else:
+        raise TypeError(f"zoom_level not understood: {type(zoom_level)}")
+    logger.debug(f"Using zoom level {zl} ({dst_res:.2f})")
+    return zl
```

Comment:
Sonar cloud remarks that this is duplicated from `data_adapter/rasterdataset.py` if it's no longer needed there please remove it, or if it is, could we move it to a common place like the utils?

Reply:
I was working on getting the slippy map tiling to work with the new architecture, but I found out while working that we decided to postpone to beta. So I already did some work, did not want to toss it.